url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ItemLogic.java

```diff
@@ -0,0 +1,622 @@
+			return;
+		}
+
+		List<Scroll_Pre> pres = new ArrayList<>();
+		List<Scroll_Suf> sufs = new ArrayList<>();
+
+		String[] prefixes = prefix.split(",");
+		for (String x : prefixes) {
+			try {
+				pres.add(StaticScrolls.searchPre(x.trim()));
+			} catch (Exception e) {
+			}
+		}
+
+		String[] suffixes = suffix.split(",");
+		for (String x : suffixes) {
+			try {
+				sufs.add(StaticScrolls.searchSuf(x.trim()));
+			} catch (Exception e) {
+			}
+		}
+
+		duplicateItems(items, false, pres, sufs);
+	}
+
+	private void duplicateItems(Items items, boolean keepChanged,
+			List<Scroll_Pre> pres, List<Scroll_Suf> sufs) {
+
+		List<Item> newitems = new ArrayList<>();
+		System.out.println(pres + " " + sufs);
+
+		if (pres.size() > 0 && sufs.size() > 0) {
+			for (Scroll_Pre pre : pres) {
+				for (Scroll_Suf suf : sufs) {
+					newitems.addAll(addScrollsToItems(items, pre, suf));
+				}
+			}
+		}
+
+		if (pres.size() > 0 && sufs.size() == 0) {
+			for (Scroll_Pre pre : pres) {
+				newitems.addAll(addScrollsToItems(items, pre, null));
+
+			}
+		}
+		if (pres.size() == 0 && sufs.size() > 0) {
+			for (Scroll_Suf suf : sufs) {
+				newitems.addAll(addScrollsToItems(items, null, suf));
+			}
+
+		}
+
+		if (!keepChanged) {
+			deleteChanged(items);
+		}
+
+		for (Item x : newitems) {
+			if (x instanceof Shoes) {
+				items.getShoes().add((Shoes) x);
+			}
+			if (x instanceof Pants) {
+				items.getPants().add((Pants) x);
+			}
+			if (x instanceof Armor) {
+				items.getArmors().add((Armor) x);
+			}
+			if (x instanceof Gloves) {
+				items.getGloves().add((Gloves) x);
+			}
+			if (x instanceof Helm) {
+				items.getHelms().add((Helm) x);
+			}
+
+		}
+
+	}
+
+	private List<Item> addScrollsToItems(Items items, Scroll_Pre newpre,
+			Scroll_Suf newsuf) {
+		List<Item> newitems = new ArrayList<>();
+
+		for (Shoes x : items.getShoes()) {
+			Shoes changed = new Shoes(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				shoes_changed.add(x);
+			}
+		}
+		for (Pants x : items.getPants()) {
+			Pants changed = new Pants(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				pants_changed.add(x);
+			}
+		}
+		for (Gloves x : items.getGloves()) {
+			Gloves changed = new Gloves(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				gloves_changed.add(x);
+			}
+		}
+		for (Armor x : items.getArmors()) {
+			Armor changed = new Armor(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				armors_changed.add(x);
+			}
+		}
+		for (Helm x : items.getHelms()) {
+			Helm changed = new Helm(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				helms_changed.add(x);
+			}
+		}
+
+		return newitems;
+	}
+
+	public void filterSetNames(Items items, String ignore) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 187-194 into a separate method.